### PR TITLE
(bond) feat: support celo for bonding products

### PR DESCRIPTION
**Pull request by Tanya-atatakai**
### 1) adds path for celo-olas

image
image

### 2) supports celo chain on bonding products
Note: there are no celo products available atm, so I tested by changing config for current OLAS-WETH pair, got almost the same value - 146, when on prod it's 146.18

Screenshot 2024-09-09 at 19 44 10

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: apps/bond/common-util/functions/web3.js [hunk 1]
```diff
@@ -1,17 +1,17 @@
+import { createPublicClient, getContract as getWagmiContract, http } from 'viem';
```
Comment (truemiller) [inline apps/bond/common-util/functions/web3.js:1]:
```suggestion
import { createPublicClient, getContract as getViemContract, http } from 'viem';
```
nitpicking here..  but wagmi just provides the hooks, viem provides the contract instance

File: apps/bond/components/BondingProducts/Bonding/BondingList/useBondingList.jsx [hunk 2]
```diff
@@ -216,15 +252,10 @@ const useAddCurrentLpPriceToProducts = () => {
...
-            //   currentLpPricePromiseList.push(currentLpPricePromise);
-            // } else
-            if (dex === DEX.BALANCER) {
+            if (dex === DEX.UNISWAP) {
+              currentLpPrice = getCurrentPriceUniswap(productList[i].token);
+              otherRequests[i] = currentLpPrice;
+            } else if (dex === DEX.BALANCER) {
               currentLpPrice = getCurrentPriceBalancer(productList[i].token);
               otherRequests[i] = currentLpPrice;
             } else if (dex === DEX.SOLANA) {
```
Comment (truemiller) [inline apps/bond/components/BondingProducts/Bonding/BondingList/useBondingList.jsx:261]: we should refactor the `DEX.###` keys at a later point..

using `ABI.UNISWAP_V2` for example, same for `.SOLANA`, would be better to identify the project and the type of contract we're interacting with

**Review by truemiller — APPROVED** [review]
looks good,
only nit pick comments, can dismiss for now

File: apps/bond/components/BondingProducts/Bonding/BondingList/utils.js [hunk 3]
```diff
@@ -59,6 +60,10 @@ export const getSvmCalculatedPriceLp = (reserveOlas, totalSupply) => {
  * Function to get the link to the LP token
  */
 export const getLpTokenLink = ({ lpDex, lpChainId, lpPoolId, productName }) => {
+  if (lpChainId === celo.id) {
```
[the remark this answers is not in the document]
Comment (Tanya-atatakai) [reply apps/bond/components/BondingProducts/Bonding/BondingList/utils.js:63]: this one is correct, but should be similar in `getCurrentPriceLpLink`, thanks! updated

**Review by kupermind — APPROVED** [review]

**Pull request merged**